x/evm/types/command.go

```diff
@@ -138,6 +138,25 @@ func NewApproveContractCallCommand(
 	}
 }
 
+// NewApproveContractCallCommandFromGeneralMessage creates a command to approve contract call
+func NewApproveContractCallCommandFromGeneralMessage(
+	chainID sdk.Int,
+	keyID multisig.KeyID,
+	msg nexus.GeneralMessage,
+) Command {
+	contractAddress := common.HexToAddress(msg.Receiver)
+	dummyTxID := common.BytesToHash(make([]byte, common.HashLength))
```

Comment:
Why use a dummy tx id instead of the actual tx id on axelar when the general msg is submitted?

Reply:
I would prefer to use the actual tx ID, but we would need to add another field to the `GeneralMessage` struct to pass this along. I also am not sure if there is or always will be a tx ID available for all general messages - though I guess in that case we could then just use a dummy tx ID.

Reply:
if we have time, we can refactor after haiyi's PR to add a separate TxID field